cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/transformers/CgmesPhaseTapChangerBuilder.java

```diff
@@ -62,7 +63,21 @@ protected void addSteps() {
         if (isLinear()) {
             addStepsLinear();
         } else if (isTabular()) {
-            addStepsFromTable();
+            PropertyBags table = context.phaseTapChangerTable(tableId);
+            if (table == null) {
+                addStepsLinear();
+                return;
+            }
+            int min = table.stream().map(p -> p.asInt(CgmesNames.STEP)).min(Integer::compareTo).orElseThrow(() -> new PowsyblException("Should at least contain one step"));
+            for (int i = min; i < min + table.size(); i++) {
+                int index = i;
+                if (table.stream().noneMatch(p -> p.asInt(CgmesNames.STEP) == index)) {
+                    context.ignored("PhaseTapChanger table", () -> String.format("There is at least one missing step (%s) in table %s. Phase tap changer considered linear", index, tableId));
+                    addStepsLinear();
+                    return;
+                }
+            }
+            addStepsFromTable(table);
```

Comment:
When the tapChanger is built the lowTapPosition is set to the lowStep (AbstractCgmesTapChangerBuilder class)

   ```
protected TapChanger build() {
        int lowStep = p.asInt(CgmesNames.LOW_STEP);
        int highStep = p.asInt(CgmesNames.HIGH_STEP);
        int neutralStep = p.asInt(CgmesNames.NEUTRAL_STEP);
        int normalStep = p.asInt(CgmesNames.NORMAL_STEP, neutralStep);
        int position = initialTapPosition(normalStep);
        if (position > highStep || position < lowStep) {
            position = neutralStep;
        }
        tapChanger.setLowTapPosition(lowStep).setTapPosition(position);

```
so to be consistent we should take `min = lowStep` instead of obtaining it from the table.

Maybe we can use a common method, something  like `isTableValid`, defined in the AbstractCgmesTapChangerBuilder class for both, `RatioTapChanger `and `PhaseTapChanger`.

Reply:
I agree with you but as in quality requirements, the table has priority over the description by attributes, we should obtain it from the table in inconsistent files. However, I will refactor a bit as you said since the code seems alike indeed.